pkg/controller/che/che_controller.go

```diff
@@ -751,138 +729,52 @@ func (r *ReconcileChe) Reconcile(request reconcile.Request) (reconcile.Result, e
 		}
 	}
 
-	addRegistryRoute := func(registryType string) (string, error) {
-		registryName := registryType + "-registry"
+	// Create devfile registry resources unless an external registry is used
+	devfileRegistryURL := instance.Spec.Server.DevfileRegistryUrl
+	externalDevfileRegistry := instance.Spec.Server.ExternalDevfileRegistry
+	if !externalDevfileRegistry {
+		registryName := "devfile-registry"
 		host := ""
 		if !isOpenShift {
-			ingress := deploy.NewIngress(instance, registryName, registryName, 8080)
-			if err := r.CreateNewIngress(instance, ingress); err != nil {
-				return "", err
+			ingressStatus := deploy.SyncIngressToCluster(instance, registryName, registryName, 8080, clusterAPI)
```

Comment:
For ingresses vs routes, it seems that for every component it is the same logic: `is (!isOpenshift) {} else {}`
Do you think we could extract that into a function (`syncRouting` or something like that ?)

Reply:
Agree. but maybe in the next PR. ^(

Reply:
OK :-)